From eb76a59139c5ca6355cf1ef9594dab8b73b9b1f5 Mon Sep 17 00:00:00 2001
Message-Id: <eb76a59139c5ca6355cf1ef9594dab8b73b9b1f5.1474443269.git.maciej.borzecki@rndity.com>
In-Reply-To: <cover.1474443269.git.maciej.borzecki@rndity.com>
References: <cover.1474443269.git.maciej.borzecki@rndity.com>
From: Maciek Borzecki <maciek.borzecki@gmail.com>
Date: Tue, 20 Sep 2016 21:24:45 +0200
Subject: [PATCH 1/2] nss: install nss modules into /lib

NSS modules (libnss_*.so.*) need to be installed into
${rootlibdir} (typically /lib) in order to be used. Previously, the
modules were installed into ${libdir} (usually ${prefix}/lib), where in
case of a split /usr, this would typically resolve to /usr/lib.

Signed-off-by: Maciek Borzecki <maciek.borzecki@gmail.com>
Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>

---
Upstream-Status: Pending

 Makefile.am | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index dbd3386..0e2dca5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -151,6 +151,7 @@ nodist_zshcompletion_DATA = $(nodist_zshcompletion_data)
 endif
 udevlibexec_PROGRAMS =
 gperf_gperf_sources =
+rootlib_LTLIBRARIES =
 
 in_files = $(filter %.in,$(EXTRA_DIST))
 in_in_files = $(filter %.in.in, $(in_files))
@@ -4931,7 +4932,7 @@ libnss_myhostname_la_LDFLAGS = \
 libnss_myhostname_la_LIBADD = \
 	libsystemd-internal.la
 
-lib_LTLIBRARIES += \
+rootlib_LTLIBRARIES += \
 	libnss_myhostname.la
 endif
 
@@ -5030,7 +5031,7 @@ libnss_mymachines_la_LDFLAGS = \
 libnss_mymachines_la_LIBADD = \
 	libsystemd-internal.la
 
-lib_LTLIBRARIES += \
+rootlib_LTLIBRARIES += \
 	libnss_mymachines.la
 
 endif
@@ -5321,7 +5322,7 @@ libnss_resolve_la_LIBADD = \
 	libsystemd-internal.la \
         -ldl
 
-lib_LTLIBRARIES += \
+rootlib_LTLIBRARIES += \
 	libnss_resolve.la
 
 systemd_resolve_SOURCES = \
-- 
2.5.0

